src/main/java/org/candlepin/model/ContentOverrideCurator.java

```diff
@@ -93,13 +107,24 @@ public void removeByParent(Parent parent) {
 
     public T retrieve(Parent parent, String contentLabel, String name) {
         if (parent != null && contentLabel != null && name != null) {
-            return (T) currentSession()
-                .createCriteria(this.entityType())
-                .add(Restrictions.eq(parentAttr, parent))
-                .add(Restrictions.eq("contentLabel", contentLabel))
-                .add(Restrictions.eq("name", name.toLowerCase()))
-                .setMaxResults(1)
-                .uniqueResult();
+            CriteriaBuilder cb = getEntityManager().getCriteriaBuilder();
+            CriteriaQuery<T> cq = cb.createQuery(this.entityType());
+            Root<T> root = cq.from(this.entityType());
+
+            Predicate parentPredicate = cb.equal(root.get(parentAttr), parent);
+            Predicate contentLabelPredicate = cb.equal(root.get("contentLabel"), contentLabel);
+            Predicate namePredicate = cb.equal(cb.lower(root.get("name")), name.toLowerCase());
+
+            cq.where(cb.and(parentPredicate, contentLabelPredicate, namePredicate));
+
+            try {
+                return getEntityManager().createQuery(cq)
```

Comment:
On this line and in most of the updates in this class, we're doubly fetching the entity manager rather than getting it once and then reusing it.

Reply:
done